### PR TITLE
feat: add sub commands '-E, --external' for 'kclvm_cli run' to specify the external package path.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #491 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm_cli

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

add sub commands '-E, --external' for 'kclvm_cli run' to specify the external package path.

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/config/src/settings.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ pub struct Config {
     pub disable_none: Option<bool>,
     pub verbose: Option<u32>,
     pub debug: Option<bool>,
+    pub package_maps: Option<HashMap<String, String>>,
```
Comment (Peefy) [inline kclvm/config/src/settings.rs:59]: More documents to explain this field `package_maps`. Why, what and how.

**Review by Peefy — COMMENTED** [review]
After merging #490, rebase the main branch and move `hashmaps_from_matches` to the `cmd::util` module

File: kclvm/config/src/settings.rs (hunk 1 above)
Comment (zong-zhe) [reply kclvm/config/src/settings.rs:61]: fixed.

File: kclvm/cmd/src/util.rs [hunk 2]
```diff
@@ -10,6 +11,28 @@ pub(crate) fn strings_from_matches(matches: &ArgMatches, key: &str) -> Option<Ve
     })
 }
 
+#[inline]
+pub(crate) fn hashmaps_from_matches(
```
Comment (Peefy) [inline kclvm/cmd/src/util.rs:15]: Failure to handle abnormal situations. For example:
https://github.com/KusionStack/KCLVM/blob/main/kclvm/driver/src/arguments.rs#L7

Comment (zong-zhe) [reply kclvm/cmd/src/util.rs:18]: fixed.

File: kclvm/runner/src/runner.rs [hunk 3]
```diff
@@ -23,6 +25,7 @@ pub type kclvm_value_ref_t = std::ffi::c_void;
 pub struct ExecProgramArgs {
     pub work_dir: Option<String>,
     pub k_filename_list: Vec<String>,
+    pub package_maps: HashMap<String, String>,
```
Comment (Peefy) [inline kclvm/runner/src/runner.rs:28]: Add `#[serde(skip)]` and `// -E key=value`

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**